src/factories/meta/SubDivideCollection_factory.h

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+// Copyright (C) 2024 Simon Gardner
+
+#pragma once
+
+#include "algorithms/meta/SubDivideCollection.h"
+#include "extensions/jana/JOmniFactory.h"
+#include "services/geometry/dd4hep/DD4hep_service.h"
+
+
+namespace eicrecon {
+
+template <class T>
+class SubDivideCollection_factory : public JOmniFactory<SubDivideCollection_factory<T>, SubDivideCollectionConfig<T>> {
+
+  public:
+    using AlgoT = eicrecon::SubDivideCollection<T>;
+
+  private:
+
+    std::unique_ptr<AlgoT> m_algo;
+
+    typename JOmniFactory<SubDivideCollection_factory<T>, SubDivideCollectionConfig<T>>::template PodioInput<T> m_raw_hits_input {this};
+    typename JOmniFactory<SubDivideCollection_factory<T>, SubDivideCollectionConfig<T>>::template VariadicPodioOutput<T> m_split_hits_output {this};
+
+    typename JOmniFactory<SubDivideCollection_factory<T>, SubDivideCollectionConfig<T>>::template Service<DD4hep_service> m_geoSvc {this};
+
+public:
+    void Configure() {
+      m_algo = std::make_unique<AlgoT>(this->GetPrefix());
+      m_algo->applyConfig(this->config());
+      m_algo->init(this->logger());
+    }
+
+    void setFunctions(std::function<std::vector<int>(const T&)> f) {
+      m_algo->setFunction(f);
+    }
+
+    void ChangeRun(int64_t run_number) {
+    }
+
+    void Process(int64_t run_number, uint64_t event_number) {
+
+    try {
+      const typename T::collection_type* raw_hits = m_raw_hits_input();
+      std::vector<gsl::not_null<typename T::collection_type*>> split_hits;
+      for (const auto& split_hit : m_split_hits_output()) {
+        split_hits.push_back(gsl::not_null<typename T::collection_type*>(split_hit.get()));
+      }
```

Comment:
Why do we need this? Cannot `algo` accept `m_split_hits_output()` as is?

Reply:
Looks like it can accept the `m_raw_hits_input()` directly but the compiler complains `m_split_hits_output()` returns a 
`std::vector<std::unique_ptr<edm4eic::RawTrackerHitCollection>>` rather than a `std::tuple<const std::vector<gsl::not_null<edm4eic::RawTrackerHitCollection*>>>`.

Reply:
I think https://github.com/eic/EICrecon/blob/main/src/global/reco/ReconstructedElectrons_factory.h does something similar, right?

Reply:
> I think https://github.com/eic/EICrecon/blob/main/src/global/reco/ReconstructedElectrons_factory.h does something similar, right?

I believe the difference will be that this hasn't been converted to an algorithms::Algorithm yet. This was the same for the templated variadic input for the collection collector. Maybe I need to better define the algorithms::Input/Output but so far nothing I've tried has worked

Reply:
FYI, https://github.com/eic/EICrecon/pull/1313/files#diff-7f55d43dee21b064cb75128d7539847e30667cd536765794d41bf7abdb32ff5fR43 is to deal with the `gsl::not_null`.

Reply:
Is that doing anything fundamentally different or just a bit neater?

Reply:
No difference. I hadn't looked at your recent changed to the code, and was just commenting on the comment https://github.com/eic/EICrecon/pull/1251#discussion_r1505700489